graphql/resolvers.go

```diff
@@ -357,6 +359,49 @@ func (r *queryResolver) TaskFiles(ctx context.Context, taskID string) ([]*Groupe
 	return groupedFilesList, nil
 }
 
+func (r *mutationResolver) SetTaskPriority(ctx context.Context, taskID string, priority int) (*restModel.APITask, error) {
+	t, err := task.FindOneId(taskID)
+	if err != nil {
+		return nil, ResourceNotFound.Send(ctx, err.Error())
+	}
+	if t == nil {
+		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("unable to find task %s", taskID))
+	}
+	authUser := gimlet.GetUser(ctx)
+	if priority > evergreen.MaxTaskPriority {
+		requiredPermission := gimlet.PermissionOpts{
+			Resource:      t.Project,
+			ResourceType:  "project",
+			Permission:    evergreen.PermissionTasks,
+			RequiredLevel: evergreen.TasksAdmin.Value,
+		}
+		isTaskAdmin := authUser.HasPermission(requiredPermission)
+		if !isTaskAdmin {
+			return nil, Forbidden.Send(ctx, fmt.Sprintf("Insufficient access to set priority %v, can only set priority less than or equal to %v", priority, evergreen.MaxTaskPriority))
+		}
+	}
+	if err = t.SetPriority(int64(priority), authUser.Username()); err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error setting task priority %v: %v", taskID, err.Error()))
+	}
+	t, err = task.FindOneId(taskID)
```

Comment:
I think we should be use the connector for most requests like this. 

I'm a little worried about calling directly into the models package so much (ever?)

Reply:
What worries you about calling into the the models package?